src/marker.js

```diff
@@ -126,7 +126,26 @@ function applyMarkersColor(path, {markerStart, markerMid, markerEnd, stroke}, st
     };
   }
 
-  if (markerStart) path.attr("marker-start", applyMarker(markerStart));
-  if (markerMid) path.attr("marker-mid", applyMarker(markerMid));
-  if (markerEnd) path.attr("marker-end", applyMarker(markerEnd));
+  if (!(markerStart || markerMid || markerEnd)) return;
+
+  const start = markerStart && applyMarker(markerStart);
+  const mid = markerMid && applyMarker(markerMid);
+  const end = markerEnd && applyMarker(markerEnd);
+  if (Z) {
+    for (const g of group(
+      path.filter((i) => i.length >= 2),
+      (d) => Z[select(d).datum()[0]]
+    ).values()) {
+      if (start) select(g.at(0)).attr("marker-start", start);
```

Comment:
Array.at is still too bleeding edge, in my opinion. (We’ve had problems with it crashing the site before, and it’s so minimally helpful over array[index] that I have difficulty justifying using it.)

https://caniuse.com/mdn-javascript_builtins_array_at